File: epi_judge_java/epi/ClosestIntSameWeight.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class ClosestIntSameWeight {

    @EpiTest(testDataFile = "closest_int_same_weight.tsv")
    public static long closestIntSameBitCount(long x) {
        for (int i = 0; i < 63; i++) {
            if (((x >> i) & 1) != ((x >> (i + 1)) & 1)) {
                x = SwapBits.swapBits(x, i, i + 1);
                break;
            }
        }

        return x;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "ClosestIntSameWeight.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/closest_int_same_weight.cc
#include "test_framework/generic_test.h"

unsigned long long ClosestIntSameBitCount(unsigned long long x) {
    int index = 0;
    while (((x >> index) & 1) == ((x >> (index + 1)) & 1)) {
        index++;
    }
    return x ^ (1LL << index) ^ (1LL << (index + 1));
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"x"};
    return GenericTestMain(args, "closest_int_same_weight.cc",
                           "closest_int_same_weight.tsv", &ClosestIntSameBitCount,
                           DefaultComparator{}, param_names);
}
